source/common/http/http1/conn_pool.cc

```diff
@@ -127,15 +127,22 @@ void ConnPoolImpl::onConnectionEvent(ActiveClient& client, Network::ConnectionEv
     // The client died.
     ENVOY_CONN_LOG(debug, "client disconnected, failure reason: {}", *client.codec_client_,
                    client.codec_client_->connectionFailureReason());
+
+    if (event == Network::ConnectionEvent::RemoteClose) {
```

Comment:
Can you potentially factor this out into a shared utility function? If you search the code IIRC this can also be pulled out from the tcp_proxy code.

Reply:
I extracted these methods to `source/common/upstream/upstream_impl.h`. If you know better place to move these methods I'll be glad to change it.